lib/helpers/errors/generic.js

```diff
@@ -0,0 +1,26 @@
+'use strict'
+
+class GenericError extends Error {
+  constructor (message, { code = 'ERR_GENERIC', details } = {}) {
+    super(message)
+
+    this.status = 500
+    this.code = code
+
+    if (details) {
+      this.details = details
+    }
+
```

Comment:
Just suggestion, I usually add also error name to be same as classname and modify stack trace and toString methods to include additional data at the beginning, e.g. like this:
```javascript
constructor (message, { code = 'ERR_GENERIC', details } = {}) {
   //...
   this.name = this.constructor.name
   //...
   Error.captureStackTrace(this)
   this.stack = `CODE: ` + this.code + (this.details ? ', DETAILS: ' + this.details : '')  + `, ${this.stack}`
}

toString () {
    return `CODE: ` + this.code + (this.details ? ', DETAILS: ' + this.details : '')  + `, ${super.toString()}`
}
```

Reply:
agree, I'll add that